content/wasm-languages/zig.md

```diff
@@ -27,85 +27,150 @@ For the most part, write your Zig code as usual. When compiling, use the target
 
 Things we like:
 
-- The Zig compiler is so nice, we now use it to compile our C as well as our Zig.
+- The Zig compiler is so nice that we now use it to compile our C and Zig.
 
 Things we're not big fans of:
 
 - When we get Wasm-related errors, they can be really terse.
 
-## Example
+# Examples
 
->> All of our examples follow [a documented pattern using common tools](/wasm-languages/about-examples).
+>> All of our examples follow [a documented pattern using common tools](/wasm-languages/about-examples). In these examples, we initially run our applications using the [wasmtime](https://github.com/bytecodealliance/wasmtime) runtime and then progress and demonstrate how to run Zig applications using the [Spin framework](../spin/v2/index).
 
-You must have [Zig](https://ziglang.org/learn/) installed.
+You must have [Zig](https://ziglang.org/learn/) installed. If you haven't already done so, please also go ahead and [install Spin](../spin/v2/install).
+
+## Example 1
+
+Below is a WebAssembly System Interface (WASI) example where Zig uses the standard library to read command line arguments.
 
 Create a new Zig program:
 
-```console
+```bash
 $ mkdir hello-zig
 $ cd hello-zig
 $ zig init-exe
+info: Created build.zig
+info: Created src/main.zig
+info: Next, try `zig build --help` or `zig build run`
 ```
 
-Inside of the `src/` directory is a file named `main.zig`. Edit it as follows:
+Inside the `src/` directory is a file named `main.zig`. Edit it as follows:
 
-```zig
+```
 const std = @import("std");
 
 pub fn main() !void {
-    const stdout = std.io.getStdOut().writer();
-    try stdout.print("content-type: text/plain\n\n", .{});
-    try stdout.print("Hello, World!\n", .{});
+    var general_purpose_allocator = std.heap.GeneralPurposeAllocator(.{}){};
+    const gpa = general_purpose_allocator.allocator();
+    const args = try std.process.argsAlloc(gpa);
+    defer std.process.argsFree(gpa, args);
+
+    for (args, 0..) |arg, i| {
+        std.debug.print("{}: {s}\n", .{ i, arg });
+    }
 }
 ```
 
-Now compile the program:
+We can now build and run the WASI example:
 
-```console
-zig build-exe -O ReleaseSmall -target wasm32-wasi src/main.zig
+```bash
+$ zig build-exe src/main.zig -target wasm32-wasi
 ```
 
-You can verify that it runs using `wasmtime`:
+As mentioned above, we are going to run our application using the [wasmtime](https://github.com/bytecodealliance/wasmtime) runtime:
 
-```console
-$ wasmtime main.wasm
-content-type: text/plain
+```bash
+$ wasmtime src/main.wasm 123 hello
+0: main.wasm
+1: 123
+2: hello
+```
+
+## Example 2
+
+In this example we are going to build a "hello world" WAGI application using Zig. We will then run the Wasm binary with wasmtime to view the output in the console, and then run it again as a [Spin](../spin/v2/index) application served over HTTP.
```

Comment:
Is there a reason to use a `..` path rather than a `/` path?  Our experience has been that `..` paths are more prone to getting messed up when we reorganise files (whereas `/` paths usually at least hit a redirect even if stuff has moved).

Reply:
It is mainly that Visual Studio Code responds to the `../` and provides a path that I can tab into (which makes it feel real[er])